Resources/MessageQueue/StompMQConnection.py

```diff
@@ -0,0 +1,172 @@
+from DIRAC import S_OK, S_ERROR
+from DIRAC.Core.Utilities.DErrno import EMQNOM, EMQUKN, EMQCONN
+
+class StompMQConnection( MQConnection ):
+  """
+  Class for management of Stomp connections
+  Allows to both send and receive messages from a queue
+  The class also implements callback functions to be able to act as a listener and receive messages
+  """
+
+  MANDATORY_PARAMETERS = [ 'Host', 'MQType' ]
+
+  def __init__( self ):
+
+    super( StompMQConnection, self ).__init__()
+    self.queueName = None
+
+    self.callback = None
+    self.receiver = False
+    self.acknowledgement = False
+    self.subscriptionID = None
+    self.msgList = []
+    self.lock = threading.Lock()
+    self.connection = None
+
+  # Callback functions for message receiving mode
+
+  def on_message( self, headers, message ):
+    """
+    Callback function called upon receiving a message
+
+    :param dict headers: headers of the MQ message
+    :param json message: json string of the message
+    """
+    result = self.callback( headers, message )
+    if self.acknowledgement:
+      if result['OK']:
+        self.connection.ack( headers['message-id'], self.subscriptionID )
+      else:
+        self.connection.nack( headers['message-id'], self.subscriptionID )
+
+  def defaultCallback( self, headers, message ):
+    """
+    Default callback function called every time something is read from the queue
+
+    :param dict headers: headers of the MQ message
+    :param json message: json string of the message
+    """
+    dictionary = json.loads( message )
+    with self.lock:
+      self.msgList.append( dictionary )
+    return S_OK()
+
+  # Rest of functions
+
+  def setupConnection( self, parameters = None, receive = False, messageCallback = None ):
+    """
+    Establishes a new connection to a Stomp server, e.g. RabbitMQ
+
+    :param dict parameters: dictionary with additional MQ parameters if any
+    :param bool receive: flag to enable the MQ connection for getting message
+    :param func messageCallback: function to be called when a new message is received from the queue ( only receiver mode ).
+                                If None, the defaultCallback method is used instead
+    :return: S_OK/S_ERROR
+    """
+
+    self.receiver = receive
+    if self.receiver:
+      if messageCallback:
+        self.callback = messageCallback
+      elif not self.callback:
+        self.callback = self.defaultCallback
+
+    if parameters is not None:
+      self.parameters.update( parameters )
+
+    # Make the actual connection
+    host = self.parameters.get( 'Host' )
+    port = self.parameters.get( 'Port', 61613 )
+    vhost = self.parameters.get( 'VHost' )
+    user = self.parameters.get( 'User', 'guest' )
+    password = self.parameters.get( 'Password', 'guest' )
+    self.queueName = self.parameters.get( 'Queue' )
+    headers = {}
+    if "Persistent" in self.parameters and self.parameters['Persistent'].lower() in ['true', 'yes', '1']:
```

Comment:
Also use `if self.parameters.get( 'Persistent', '' ).lower() in [ 'true', 'yes', '1' ]:` here?